central/detection/lifecycle/manager_impl.go

```diff
@@ -230,25 +235,68 @@ func (m *managerImpl) addBaseline(deploymentID string) {
 			ProtoQuery(),
 	)
 
-	m.buildMapAndCheckBaseline(indicatorSlice)
+	baselines := m.buildMapAndCheckBaseline(indicatorSlice)
+
+	userLock := features.AutoLockProcessBaselines.Enabled()
```

Comment:
If I'm following, this is only called from flush baseline meaning observation is over.  

So if a user locks and unlocks a baseline during observation.  We will re-lock it.  Seems like an odd and likely rare case for someone using this feature so probably not a big deal.

If a user unlocks after observation is over we will NOT re-lock it when we process additional indicators which was going to be a problem with putting the logic in `checkAndUpdateBaseline`.

Reply:
That is correct. I think this strategy is a lot simpler as we don't need to worry in `checkAndUpdateBaseline` if the process indicators or process baselines queue is being flushed, and we don't need to modify every instance of database updates in that function, or worry if it is returning too early.

There is a corner case that you mentioned of someone locking and unlocking a process baseline, before the observation period is over. That will likely be rare, and I can address that case in a different PR. I don't yet know how I would do that.

Reply:
I would address that corner case by writing a ticket on the epic of low priority and deal with it only if we have to.  Especially since auto locking will eventually be added to cluster config and not just a blanket feature flag.

Reply:
I have created the ticket here https://issues.redhat.com/browse/ROX-30688